Fix odd scenario where Docker Compose fails to create brand new databases

tahb opened the pull request:
<!-- Do you need to update the changelog? -->

## Changes in this PR

We ran into this problem on CAEW https://github.com/citizensadvice/corporate-api/pull/128/commits/8c5e0c1f6da6ebaec70dd5488c10e654e88fe4e1. We aren't sure how this worked before without this configuration?

tahb committed:
CI should never run with postgres state from any previous runs

tahb committed:
Docker Compose creates postgres containers safely initialise the DB

POSTGRES_DB
---
This optional environment variable can be used to define a different name for the default database that is created when the image is first started. If it is not specified, then the value of POSTGRES_USER will be used.[1]

Before, when we were providing a POSTGRES_USER we used the value of "postgres".  This instructed postgres to create a database that did not match the name app-test or rails-template-development (depending on your environment). For some reason, this worked for a time but we noticed that throwing away all local database state within volumes, we got a database not found error on boot for any environment.

This change explicitly instructs Postgres as to the name of the database we expect it to create. It should match the value we use at the end of DATABASE_URL.

[1] https://hub.docker.com/_/postgres/

mec commented:
I am not clear on `docker-compose.test.yml` and `docker-compose.ci.yml` and the changes here, maybe it is the commit messages?

tahb commented:
@mec fair challenge 👍 

I think we are in a tricky and uninformed spot. We can't tell exactly how we got into the state where we had a postgres docker container running without an initialised database.

Throwing away local volumes and state, changing the database names and jumping between different versions of postgres - none of those steps get us back into a failing state.

My thought process is that whilst we don't know why things broke, passing this value to our Postgres containers shouldn't risk any new failures but should protect us from this odd situation in future. I think it's applicable on dev, test or ci as in all cases there is the case where a user creates (or recreates) the database for the first time. 

(I'll improve the PR description as it's lacking this extra bit)

mec commented:
@tahb yeah no probs! The changes are good (I've seen the result!) but I just struggle to understand, I know it's painful to wait for my brain to catchup! 

On this I think it was the commit message talking about CI but the commit changing the `docker-compose.test.yml` file?

tahb commented:
Ah I did do "Docker in CI and test" but totally neglected to expand in the description as to what that means because of course on CI we do tests, and "test" is overused!

I might propose this for all 3 docker-compose files and improve the description of that commit.

tahb committed:
Docker Compose is explicit about the Postgres user name

`postgres` is the default [1] so this should be a refactoring that makes each docker compose file dev, test and CI, the same.

[1] https://hub.docker.com/_/postgres/

tahb committed:
Docker Compose for CI can be run locally in conjunction with test

By using the same network name: "test" for both test and ci compose files, we run the risk of one being able to connect to services provisioned by the other.

Rename the network name to rule that possibility out.

tahb commented:
@mec I've done some tinkering. Ready for your eyeballs again!

tahb commented on docker-compose.yml:
```diff
@@ -29,6 +29,8 @@ services:
     volumes:
       - db-data:/var/lib/postgresql/data
     environment:
+      # TODO: Update rails-template with application name
```
We have the comment about so it felt right to copy it here too. Ideally all the databases have relevant names so you can more easily see what you have running when switching between projects. Another day I think?

mec reviewed (approved):
Loads better! Thanks for bearing with me!

CristinaRO commented:
@tahb This check is stuck: `LGTM analysis: JavaScript Expected — Waiting for status to be reported`. Do you want to look into it or override and merge?

tahb commented:
@CristinaRO yes please! Thank you
